framework/src/media/audio/audio_manager.c

```diff
@@ -1429,6 +1434,8 @@ audio_manager_result_t find_stream_in_device_with_process_type(device_process_ty
 
 	process_type = get_process_type_audio_param_value(type);
 	subprocess_type = get_subprocess_type_audio_param_value(subtype);
+
+	memset(&caps_desc, 0, sizeof(struct audio_caps_desc_s));
```

Comment:
I think memset is not neccessary because this is local structure.
If there are any problem relatedto cpas_desc, you should check line 1469 first.

Reply:
I don't agree with you, please think about role of caller & callee one more time.
caps_dec is kind of object to get data from driver, so it doesn't know what is default value. 
So driver should handle this.
```
static int DRIVER_getcaps(..., struct audio_caps_s *caps)
{
caps->ac_control.b[0] = AUDIO_PU_UNDEF;
if (foo) {
    caps->ac_controls.b[0] |= AUDIO_PU_SPEECH_DETECT;
}
...
}
```

Reply:
I got it. Then I'll fix the driver code

Reply:
I have a question.
What if `spi driver` control `AUDIO_TYPE_PROCESSING ac_type` case, but `i2s driver` doesn't?
Should all our driver code have codes that control all Audio Device Types?

```cpp
static int spi_getcpas()
{
    switch(caps->ac_types) {
    case AUDIO_TYPE_PROCESSING:
        caps->ac_control.b[0] = AUDIO_PU_SPEECH_DETECT;
        break;
    default:
        break;
    }
}

static int i2s_getcpas()
{
    switch(caps->ac_types) {
    /* Below case doesn't exist */
    /* But caps->ac_controls.b[0] should be set */
    //case AUDIO_TYPE_PROCESSING:
    //    break;
    default:
        break;
    }
}

```

Reply:
In AUDIOIOC_GETCAPS case,
we can control like below. How do you think?
```cpp
if (caps_desc.caps.ac_subtype == 0 && caps_desc.caps.ac_channels == 0) {
		meddbg("Unsupported ac_type\n");
		caps_desc.caps.ac_subtype = process_type;
		close(fd);
		pthread_mutex_unlock(&(card->card_mutex));
		continue;
	}
```

Reply:
I think getcaps should return -1 and set errno as EINVAL.
How do you think?

Reply:
I already defined UNDEF, please check audio.h

Reply:
Could you explain more?
I can't catch it